minio/mcs
# IDP: OpenID integration for mcs

This PR adds support for oidc in  mcs, to enable idp
authentication you need to pass the following environment variables and
restart mcs.

```
MCS_IDP_URL=""
MCS_IDP_CLIENT_ID=""
MCS_IDP_SECRET=""
MCS_IDP_CALLBACK=""
```

Make sure the mcsAdmin policy exists on minio.

## Test 1

- Create and configure a new account in auth0
- Create a new application and get the application credentials (clientId, secret, url, etc) on auth0
- Set the mcs callback on auth0
- Create a new user in auth0
- Start MCS with the idp env variables
- Click on the Login button and authenticate using your auth0 account
- You should be able to access MCS_IDP_SECRET

## Test 2

- Disable the user you just created on auth0
- Click on the Login button and authenticate using your auth0 account
- You should be unable to login because the user is disabled in auth0
